app/controllers/consent_forms/edit_controller.rb

```diff
@@ -142,14 +145,18 @@ def next_health_answer_missing_response?
   def skip_to_confirm_or_next_health_question
     if skip_to_confirm?
       if is_health_question_step? && next_health_answer_missing_response?
-        jump_to "health-question",
-                question_number: next_health_question,
-                skip_to_confirm: true
+        jump_to "health-question", question_number: next_health_question
       else
         jump_to Wicked::FINISH_STEP
       end
     elsif is_health_question_step? && next_health_question
       jump_to "health-question", question_number: next_health_question
     end
   end
+
+  def set_follow_up_changes_start_page
+    if params[:skip_to_confirm] && is_health_question_step?
```

Comment:
Is `skip_to_confirm?` a better source of truth here, or would this introduce bugs?

```suggestion
    if skip_to_confirm? && is_health_question_step?
```

Reply:
This won't work because we should only set `session[:follow_up_changes_start_page]` on the first health question that gets visited from the confirmation page. Lemme try to illustrate:

```
GET /sessions/2/consents/edit/confirm
# User clicks on the change link for the asthma question.
# This first GET is where "follow_up_changes_start_page" gets set
GET /sessions/2/consents/edit/health-question?question_number=0&skip_to_confirm=true
# User changes the answer to "Yes", causing follow-up questions to be viewed
PUT /sessions/2/consents/edit/health-question?question_number=0
# "follow_up_changes_start_page" should _not_ get set here
GET /sessions/2/consents/edit/health-question?question_number=1
```

If we make that suggested change above, then we risk setting `follow_up_changes_start_page` to "1" which would break the mechanism, unfortunately.

This does, point out though, that naming may be improved.